modules/luci-compat/luasrc/tools/webadmin.lua

```diff
@@ -42,9 +42,9 @@ function date_format(secs)
 	end
 	
 	if days > 0 then
-		return string.format("%.0fd %02.0fh %02.0fmin %02.0fs", days, hour, mins, secs)
+		return string.format("%.0f\u{202f}d %02.0f\u{202f}h %02.0f\u{202f}min %02.0f\u{202f}s", days, hour, mins, secs)
```

Comment:
The `\u{....}` syntax is not supported by Lua 5.1. You need to paste the character literally.

Reply:
That sounds like quite complication, especially at the translations stage.

I suggest that it is left as it has been.

Reply:
@jow- Done.

@hnyman The strings should not be translated, as symbols are universal. That's one of the primary motivations for changing everything to symbols.

Reply:
> @hnyman The strings should not be translated, as symbols are universal. 

Oh, they are absolutely not universal and will need to be translated if you look outside of the bubble of germanic group of languages.

If half-space is a violation of the standards system anyways, why implement it at all? This is no longer about standards, this is about an insistence that "2 h 2 m 2 s" has better readability than "2h 2m 2s". 

The half-space and no space are both violations, why not let things be as they used to be?

Until those strings are localizable and locale owners can change spacing and the order of things as most suitable for the locales, I strongly believe they should be left as is, as that had been universally accepted way of presenting information so far, based on the fact that no one has complained.

Reply:
@stangri Symbols [are universal](https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9-EN.pdf#page=10); that's why they're referred to as symbols instead of abbreviations (which are translated). There's akin to the per cent symbol (%): it's the same in all languages, doesn't change between singular/plural based on the quantity, and does not change based on upper-/lowercase. You can see it on this Japanese food package as an example:

![81229561-a519b780-8fb5-11ea-8075-a4f7b780a68b](https://user-images.githubusercontent.com/2997884/83661153-58f96d00-a58b-11ea-8d58-5252b09dd1de.png)

And the half-space is not a violation per se; the standard calls for [_some_ kind of space](https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9-EN.pdf#page=35), but doesn't specify the width. It is customary to use a thin space as the thousands separator and a full space to separate the quantity and the unit symbol.

The other main focus was about "m" being the symbol for the metre, not for minute. The symbol for the minute is "min", which is one of the main changes. Something does not become universally accepted just because no one has complained about it, it just means they have tolerated it or might not have noticed it. This is not to mention the fact that this PR is implicitly a complaint about the very issue.

Reply:
> @stangri Symbols [are universal](https://www.bipm.org/utils/common/pdf/si-brochure/SI-Brochure-9-EN.pdf#page=10); that's why they're referred to as symbols instead of abbreviations (which are translated). They're akin to the per cent symbol (%): it's the same in all languages, doesn't change between singular/plural based on the quantity, and does not change based on upper-/lowercase. 

No, it's definitely NOT similar to the % symbol and NOT the same in ALL languages, check out this wiki article about kilogramm in Russian: https://ru.wikipedia.org/wiki/%D0%9A%D0%B8%D0%BB%D0%BE%D0%B3%D1%80%D0%B0%D0%BC%D0%BC. It mentiones `kg` 3 times, the rest of the article uses the russian abbreviation `кг` to refer to kilogramm (14 times), not `kg`.

While some locales may be more tolerant to using SI symbols instead of their own abbreviations, but definitely not all will be.

> The other main focus was about "m" being the symbol for the metre, not for minute. The symbol for the minute is "min", which is one of the main changes. 

I believe that "main focus" wasn't implemented until after I brought up the spacing.

We're not talking about a scientific paper, we're talking about a device (in many cases consumer device) around the world, we should use what makes sense for (in this case, no spacing and m for minutes) and is in common use by the customer/user and not force something they are not accustomed to.

Reply:
> we should use what makes sense for (in this case, no spacing and m for minutes) and is in common use by the customer/user and not force something they are not accustomed to.

+1 for stangri's comment. The old way "2h 45m 32s" is better, although not strictly grammatically correct.

Reply:
> Regardless, 96%+ of the world uses the SI as their sole system of measurement, and 100% of the world encounters the SI in some way or another on almost a daily basis.

Could you please provide reference to back that up. I'm not sure you're aware that a large portion of just the English speaking population is more accustomed to imperial units, not metric ones. Also part of that population doesn't even think in terms of 24 hours and thinks am/pm. How does that conform to SI as the sole system of measurement for 96%+?

> I don't understand the resistance here. 

Like I said before, IMHO "2 h 2 m 2 s" or even "2 h 2 min 2 s" are significantly harder to quick-read than "2h 2m 2s" and carry no benefit to the user.

Reply:
> Could you please provide reference to back that up. I'm not sure you're aware that a large portion of just the English speaking population is more accustomed to imperial units, not metric ones. Also part of that population doesn't even think in terms of 24 hours and thinks am/pm. How does that conform to SI as the sole system of measurement for 96%+?

[Sure](https://www.nist.gov/system/files/documents/pml/wmd/metric/1136a.pdf#page=3). The overwhelming majority of the [English-speaking population](https://en.wikipedia.org/wiki/English-speaking_world#English_as_a_global_language) is [accustomed to the metric system](https://www.cia.gov/library/publications/the-world-factbook/appendix/appendix-g.html), not imperial or US customary units. There might be some imperial/US customary units people still use, but those aren't the ones being discussed here. My main point is that being 96% of the population lives in a country where the metric system is the only legal measurement system or is legally the one to be used for the units we're talking about. This means that 96% of the population is accustomed to seeing "1 h 30 min", not "1h 30m", which is also not correct grammatically. This is not to mention that the rest of the 4% of the population encounters metric units almost on a daily basis, so claiming they're unaccustomed to seeing "1 h 30 min" is dubious. The 24-hour clock is not a part of the SI, and is only tangentially related to it (i.e., they usually go together, but are separate).

> Like I said before, IMHO "2 h 2 m 2 s" or even "2 h 2 min 2 s" are significantly harder to quick-read than "2h 2m 2s" and carry no benefit to the user.

I understand that you think it hinders readability, but this is purely subjective; I would disagree with that viewpoint, but that's besides the point. I don't think it's a convincing argument given at least 96% of the world disagrees with you. I sense the general argument here is that "this is how it always has been, so this is how it should be"; that argument, however, is never a strong one generally (let alone in this case), especially knowing that in this case the "way it has been done" is incorrect and not preferred by practically all of the population.